components/movementsensor/replay/replay.go

```diff
@@ -454,7 +513,79 @@ func addGRPCMetadata(ctx context.Context, timeRequested, timeReceived *timestamp
 	return nil
 }
 
-// extractDataAndMetadata retrieves the next cached data and removes it from the cache. It assumes the write lock is being held.
+func (replay *replayMovementSensor) setProperty(method method, supported bool) error {
+	switch method {
+	case position:
+		replay.properties.PositionSupported = supported
+	case linearVelocity:
+		replay.properties.LinearVelocitySupported = supported
+	case angularVelocity:
+		replay.properties.AngularVelocitySupported = supported
+	case linearAcceleration:
+		replay.properties.LinearAccelerationSupported = supported
+	case compassHeading:
+		replay.properties.CompassHeadingSupported = supported
+	case orientation:
+		replay.properties.OrientationSupported = supported
+	default:
+		return errors.New("can't set property, invalid method: " + string(method))
+	}
+	return nil
+}
+
+// attemptToInitializeProperty will try to update the cache for the provided method.
+func (replay *replayMovementSensor) attemptToInitializeProperty(ctx context.Context, method method) (bool, error) {
+	initializedProperty := false
+	if replay.closed {
+		return initializedProperty, errSessionClosed
+	}
+	if err := replay.updateCache(ctx, method); err != nil && !strings.Contains(err.Error(), ErrEndOfDataset.Error()) {
```

Comment:
[nit/opt] I'd remove the && here to make this more readable and instead have the check for ErrEndOfDataset be nested inside this if. But up to you